frontend/src/bundles/common/components/stepper/common/components/step/step.tsx

```diff
@@ -0,0 +1,66 @@
+import { Box, Flex, Text } from '../../../../components.js';
+import { selectIcon } from '../components.js';
+
+const Step = ({
+    step,
+    stepDescription,
+}: {
+    step:
+        | 'completed'
+        | 'current'
+        | 'default'
+        | 'lastIncompleted'
+        | 'lastCompleted';
+    stepDescription: string;
+}): JSX.Element => {
+    const afterStyles = {
+        zIndex: -1,
+        content: '""',
+        position: 'relative',
+        top: '1.5rem',
+        left: '50%',
+        height: '2px',
+        backgroundColor: step === 'completed' ? 'white' : '#35399a',
+        order: -1,
+    };
+
+    return (
+        <Flex
+            sx={{
+                flex: 1,
+                flexDirection: 'column',
+                textAlign: 'center',
+            }}
+            _before={{
+                content: '""',
+                position: 'relative',
+                zIndex: 1,
+                display: 'block',
+                borderRadius: '50%',
+                margin: '0.7rem auto 0',
+            }}
+            _after={
+                step === 'lastIncompleted' || step === 'lastCompleted'
+                    ? {}
+                    : afterStyles
+            }
+        >
+            <Box>{selectIcon(step)}</Box>
```

Comment:
Since selectIcon return jsx maybe it it better to adjust the code so you can use selectIcon as component? 
`<Box><SelectIcon step={step} /></Box>`
But if you decide to choose this approach I guess the name of the component should be changed, for example StepIcon.

Reply:
done